src/main.c

```diff
@@ -2176,12 +2177,129 @@ void on_middle_click() {
-    // }
+    printf("Initializing linmath code for drawing in window\n");
+    static const struct{
+        float x, y;
+        float r, g, b;
+    } 
+    vertices[3] = {
+        { -0.6f, -0.4f, 1.f, 0.f, 0.f },
+        {  0.6f, -0.4f, 0.f, 1.f, 0.f },
+        {   0.f,  0.6f, 0.f, 0.f, 1.f }
+    };
+ 
+    static const char* vertex_shader_text =
+        "#version 110\n"
+        "uniform mat4 MVP;\n"
+        "attribute vec3 vCol;\n"
+        "attribute vec2 vPos;\n"
+        "varying vec3 color;\n"
+        "void main()\n"
+        "{\n"
+        "    gl_Position = MVP * vec4(vPos, 0.0, 1.0);\n"
+        "    color = vCol;\n"
+        "}\n";
+ 
+    static const char* fragment_shader_text =
+        "#version 110\n"
+        "varying vec3 color;\n"
+        "void main()\n"
+        "{\n"
+        "    gl_FragColor = vec4(color, 1.0);\n"
+        "}\n";
+
+    GLuint vertex_buffer, vertex_shader, fragment_shader, program;
+    GLint mvp_location, vpos_location, vcol_location;
+
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MAJOR, 2);
+    glfwWindowHint(GLFW_CONTEXT_VERSION_MINOR, 0);
+
+    printf("Creating new window\n");
+    GLFWwindow* w = glfwCreateWindow(200, 200, "Pause", NULL, NULL);
+    printf("Showing pause window.\n");
+    glfwShowWindow(w);
+    printf("Now awaiting keyboard input on new window.\n");
+    glfwSetKeyCallback(w, pause_key_callback);
+
+    glGenBuffers(1, &vertex_buffer);
+    glBindBuffer(GL_ARRAY_BUFFER, vertex_buffer);
+    glBufferData(GL_ARRAY_BUFFER, sizeof(vertices), vertices, GL_STATIC_DRAW);
+ 
+    vertex_shader = glCreateShader(GL_VERTEX_SHADER);
+    glShaderSource(vertex_shader, 1, &vertex_shader_text, NULL);
+    glCompileShader(vertex_shader);
+ 
+    fragment_shader = glCreateShader(GL_FRAGMENT_SHADER);
+    glShaderSource(fragment_shader, 1, &fragment_shader_text, NULL);
+    glCompileShader(fragment_shader);
+ 
+    program = glCreateProgram();
+    glAttachShader(program, vertex_shader);
+    glAttachShader(program, fragment_shader);
+    glLinkProgram(program);
+ 
+    mvp_location = glGetUniformLocation(program, "MVP");
+    vpos_location = glGetAttribLocation(program, "vPos");
+    vcol_location = glGetAttribLocation(program, "vCol");
+ 
+    glEnableVertexAttribArray(vpos_location);
+    glVertexAttribPointer(vpos_location, 2, GL_FLOAT, GL_FALSE,
+                          sizeof(vertices[0]), (void*) 0);
+    glEnableVertexAttribArray(vcol_location);
+    glVertexAttribPointer(vcol_location, 3, GL_FLOAT, GL_FALSE,
+                          sizeof(vertices[0]), (void*) (sizeof(float) * 2));
+    
+    printf("Executing drawing code.\n");
+    while (!glfwWindowShouldClose(w))
+    {
+        float ratio;
+        int width, height;
+        //mat4x4 m, p, mvp;
+ 
+        glfwGetFramebufferSize(w, &width, &height);
+        ratio = width / (float) height;
+ 
+        glViewport(0, 0, width, height);
+        glClear(GL_COLOR_BUFFER_BIT);
+ 
+        // mat4x4_identity(m);
```

Comment:
If this code is unneeded I would suggest taking it out. If it is needed maybe just comment why extra code is left in. Besides this the code looks formatted correctly and clean. Nice work!